tests/scan/test_basic.py

```diff
@@ -240,6 +241,54 @@ def scan_nodes_from_fct(fct):
 
 
 class TestScan:
+    @pytest.mark.parametrize(
+        "rng_type",
+        [
+            np.random.default_rng,
+            np.random.RandomState,
+        ],
+    )
+    def test_inner_graph_cloning(self, rng_type):
+        r"""Scan should remove the updates-providing special properties on `RandomType`\s."""
+
+        inner_inner_rng = shared(rng_type(), name="inner_inner_rng")
+
+        y = shared(np.array(1.0, dtype=config.floatX), name="y")
+        y.default_update = y + 1
+
+        z_rng = shared(rng_type(), name="z_rng")
+        z = normal(0, 1, rng=z_rng, name="z")
+
+        z_rng_update = z.owner.outputs[0]
+        z_rng_update.name = "z_rng_update"
+        z_rng.default_update = z_rng_update
+
+        inner_rng = None
+
+        def inner_fn(x):
+            inner_rng = shared(rng_type(), name="inner_rng")
+            inner_rng.default_update = inner_inner_rng
+            inner_inner_rng.default_update = inner_rng
```

Comment:
Can you add a comment explaining why the test is setting up this circular chaining?

Reply:
I simply didn't want to create another shared RNG object just to test nested `.default_update` behavior; otherwise, the only place where that effectively matters is during the computation/evaluation, which doesn't affect the things being confirmed by this test.